### PR TITLE
patch: add default categorisation to response

### PR DESCRIPTION
**Pull request by saminahbab**
### What

Add default categorisation flag to filter flex api 
ammend tests to include default categorisation to all dimension responses

### CODE REVIEW

File: api/contract.go [hunk 1]
```diff
@@ -201,6 +202,8 @@ func (d *dimensionItem) fromDimension(dim model.Dimension, host, filterID string
 	d.Name = dim.Name
 	d.Label = dim.Label
 	d.FilterByParent = dim.FilterByParent
+	d.DefaultCategorisation = dim.DefaultCategorisation
+
```
Comment (ChrysmOre) [inline api/contract.go:206]: blank line?

File: features/multivariate_put_dimension.authorized.feature [hunk 2]
```diff
@@ -183,4 +185,3 @@ Feature: Multivariate Feature Dimensions Private Endpoints
       "total_count": 2
     }
     """
```
Comment (ChrysmOre) [inline features/multivariate_put_dimension.authorized.feature:187]: MIssing CR

File: features/filter_dimension.authorised.feature [hunk 3]
```diff
@@ -117,6 +117,7 @@ Feature: Filter Outputs Private Endpoints Enabled
           "name": "geography",
           "id": "city",
           "label": "City",
+          "default_categorisaion":  "",
```
Comment (ChrysmOre) [inline features/filter_dimension.authorised.feature:120]: Adding these blank values is no different from omitting completely. You should add a value if you want to capture any new functionality.

File: api/contract.go [hunk 4]
```diff
@@ -186,11 +186,12 @@ type updateFilterDimensionResponse struct {
...
-	ID             string             `json:"id"`
-	Name           string             `json:"name"`
-	Label          string             `json:"label"`
-	FilterByParent string             `json:"filter_by_parent,omitempty"`
-	Links          dimensionItemLinks `json:"links"`
+	ID                    string             `json:"id"`
+	Name                  string             `json:"name"`
+	Label                 string             `json:"label"`
+	FilterByParent        string             `json:"filter_by_parent,omitempty"`
+	DefaultCategorisation string             `json:"default_categorisaion"`
```
Comment (ChrysmOre) [inline api/contract.go:193]: Would we want to omitempty?

File: features/filter_dimension.authorised.feature (hunk 3 above)
Comment (saminahbab) [reply features/filter_dimension.authorised.feature:120]: Nick mentioned that he needed a clear signal for the frontend controllers to understand when there is a default categorisation, so we thought of including it always, with the empty string representing a nil value, thus no default categorisation.

File: api/contract.go (hunk 4 above)
Comment (saminahbab) [reply api/contract.go:193]: i dont think so based on the conversation me and Jit had

File: features/multivariate_put_dimension.authorized.feature (hunk 2 above)
Comment (saminahbab) [reply features/multivariate_put_dimension.authorized.feature:187]: should be fixed now? right side shows new lines added

**Review by j-s-rawat — APPROVED** [review]
LGTM

**Pull request merged**